proto/internal/temporal/server/api/persistence/v1/executions.proto

```diff
@@ -249,6 +255,41 @@ message WorkflowExecutionInfo {
     VersionedTransition signal_request_ids_last_update_versioned_transition = 94;
 
     repeated StateMachineTombstoneBatch sub_state_machine_tombstone_batches = 95;
+
+    // Holds all the information about versioning for this particular execution.
+    VersioningInfo versioning_info = 96;
+
+    message VersioningInfo {
+        // Determines how server should treat this execution when workers are upgraded.
+        // When present it means that this workflow is versioned. This is not set until an execution
+        // completes its first workflow task on a versioned worker.
+        temporal.api.enums.v1.VersioningBehavior behavior = 1;
+        // The worker deployment to which this workflow is assigned currently.
+        // Must be present if and only if `behavior` is set.
+        temporal.api.common.v1.WorkerDeployment deployment = 2;
+        // Manual override for execution's versioning behavior. Takes precedence over `behavior`.
+        temporal.api.enums.v1.VersioningBehavior behavior_override = 3;
+        // Used to manually pin the execution to a deployment. Must be set when if and only if
+        // `behavior_override` is PINNED. Takes precedence over `deployment`.
+        temporal.api.common.v1.WorkerDeployment deployment_override = 4;
+        // When present, indicates the workflow is being redirected to a different deployment.
+        // A deployment redirect can only exist during the life time of a pending workflow task.
+        // If the pending workflow task completes (at the next WorkflowTaskCompleted event), the
+        // redirect is considered complete and the workflow's deployment is updated. If the pending
+        // workflow task fails or times out, then the redirect is canceled and workflow remains on
+        // the previous deployment.
```

Comment:
> When the pending WFT completes and the workflow's deployment is updated, is the worklfow's redirect_info also deleted?
yes, it will be deleted. That's part of what I wanted to convey with the first sentence: "When present, indicates the workflow is being redirected to a different deployment.".

BTW, I ended up just removing the "deployment" out of "deployment redirect".